cdap-docs/hydrator-manual/source/developing-plugins/creating-a-plugin.rst

```diff
@@ -876,18 +738,306 @@ Example::
     }
   }
 
+.. highlight:: java
 
-Post-run Action Plugin
-======================
-In order to implement an Post-run Action plugin (to be used with the Data Pipeline
-artifact), you extend the ``PostAction`` class. Only one method is required to be
-implemented::
+Streaming Source Plugin
+=======================
+A Streaming Source plugin is used as a source in realtime data pipelines.
+It is used to fetch a Spark DStream, which is an object that represents a
+collection of Spark RDDs and that produces a new RDD every batch interval of the pipeline. 
```

Comment:
Here are links to these terms (_Spark DStream_, _Spark RDD_), or should we just assume that people know or can find out about them?

```
It is used to fetch a `Spark DStream 
<https://spark.apache.org/docs/latest/streaming-programming-guide.html#discretized-streams-dstreams>`__, 
which is an object that represents a collection of `Spark RDDs 
<https://spark.apache.org/docs/latest/programming-guide.html#resilient-distributed-datasets-rdds>`__
and that produces a new RDD every batch interval of the pipeline.